internal/monitors/elasticsearch/client/indexstatsfetcher.go

```diff
@@ -0,0 +1,472 @@
+			prepareGaugeHelper(fmt.Sprintf("gauge.indices.%s.translog.size", aggregationType), defaultDims, translog.SizeInBytes),
+		}...)
+	}
+
+	return out
+}
+
+func (requestCache *RequestCache) fetchIndexRequestCacheStats(enhanced bool, defaultDims map[string]string, aggregationType string) []*datapoint.Datapoint {
+	var out []*datapoint.Datapoint
+
+	if enhanced {
+		out = append(out, []*datapoint.Datapoint{
+			prepareCumulativeHelper(fmt.Sprintf("counter.indices.%s.request_cache.evictions", aggregationType), defaultDims, requestCache.Evictions),
+			prepareCumulativeHelper(fmt.Sprintf("counter.indices.%s.request_cache.hit_count", aggregationType), defaultDims, requestCache.HitCount),
+			prepareCumulativeHelper(fmt.Sprintf("counter.indices.%s.request_cache.miss_count", aggregationType), defaultDims, requestCache.MissCount),
+		}...)
+	}
+
+	out = append(out, []*datapoint.Datapoint{
+		prepareGaugeHelper(fmt.Sprintf("gauge.indices.%s.request_cache.memory-size", aggregationType), defaultDims, requestCache.MemorySizeInBytes),
+	}...)
+
+	return out
+}
+
+func (recovery *Recovery) fetchIndexRecoveryStats(enhanced bool, defaultDims map[string]string, aggregationType string) []*datapoint.Datapoint {
+	var out []*datapoint.Datapoint
+
+	if enhanced {
+		out = append(out, []*datapoint.Datapoint{
+			prepareGaugeHelper(fmt.Sprintf("gauge.indices.%s.recovery.current-as-source", aggregationType), defaultDims, recovery.CurrentAsSource),
+			prepareGaugeHelper(fmt.Sprintf("gauge.indices.%s.recovery.current-as-target", aggregationType), defaultDims, recovery.CurrentAsTarget),
+			prepareCumulativeHelper(fmt.Sprintf("counter.indices.%s.recovery.throttle-time", aggregationType), defaultDims, recovery.ThrottleTimeInMillis),
+		}...)
+	}
+
+	return out
+}
+
+func (idCache *IDCache) fetchIndexIDCacheStats(enhanced bool, defaultDims map[string]string, aggregationType string) []*datapoint.Datapoint {
+	var out []*datapoint.Datapoint
+
+	if enhanced {
+		out = append(out, []*datapoint.Datapoint{
+			prepareGaugeHelper(fmt.Sprintf("gauge.indices.%s.id_cache.memory-size", aggregationType), defaultDims, idCache.MemorySizeInBytes),
+		}...)
+	}
+
+	return out
+}
+
+func (suggest *Suggest) fetchIndexSuggestStats(enhanced bool, defaultDims map[string]string, aggregationType string) []*datapoint.Datapoint {
+	var out []*datapoint.Datapoint
+
+	if enhanced {
+		out = append(out, []*datapoint.Datapoint{
+			prepareGaugeHelper(fmt.Sprintf("gauge.indices.%s.suggest.current", aggregationType), defaultDims, suggest.Current),
+			prepareCumulativeHelper(fmt.Sprintf("counter.indices.%s.suggest.time", aggregationType), defaultDims, suggest.TimeInMillis),
+			prepareCumulativeHelper(fmt.Sprintf("counter.indices.%s.suggest.total", aggregationType), defaultDims, suggest.Total),
+		}...)
+	}
+
+	return out
+}
+
+func (percolate *Percolate) fetchIndexPercolateStats(enhanced bool, defaultDims map[string]string, aggregationType string) []*datapoint.Datapoint {
```

Comment:
Rename all these that start with `fetch`.  That strongly implies that the function is doing something externally to generate the metrics, but it is really just mapping the struct to datapoints.